lib/util/buffer.c

```diff
@@ -194,98 +194,52 @@ void buffer_clear( buffer_type * buffer ) {
 }
 
 
-/*****************************************************************/
-/**
-    Observe that it is the functions with _safe_ in the name which
-    most closely mimicks the behaviour of fread(), and fwrite() -
-    these functions will *NOT* abort if the buffer is to small,
-    instead they will just return the number of items read/written,
-    and it is the responsability of the calling scope to check the
-    return values.
-
-    The functions buffer_fread() and buffer_fwrite() will abort if
-    read/write to buffer failed.
-*/
-
-
-
-static size_t buffer_fread__(buffer_type * buffer , void * target_ptr , size_t item_size , size_t items, bool abort_on_error) {
+size_t buffer_fread(buffer_type * buffer,
```

Comment:
The return value of this function gives no information as the function either `util_aborts`, or returns the number of items requested. I suggest to make the function `void`.